### PR TITLE
Dual-license `error-stack` and misc drive-by updates

### PR DESCRIPTION
**Pull request by nonparibus**
## 🌟 What is the purpose of this PR?

This PR combines multiple miscellaneous changes.

- Adds mention of our CLA Assistant bot/process to the top-level `CONTRIBUTING` file.
- Adds an additional license option to our `error-stack` Rust crate (introducing Apache 2.0 as an option, complementing the existing MIT license).
- Improves license markdown formatting.
- Improves the structure of the `error-stack` README and adds additional information.
- Adds additional information to our open-source strategy blog post.
- Updates certain README text ahead of planned monorepo structure improvements.
- Updates the filenames of our markdown blog posts to enable better cross-env chronological sorting.
- Updates the titles of various blog post authors to reflect their positions at HASH.
- Improves title of 'blocks' GitHub Issue bug report form
- Adds `deer` crate GitHub Issue bug report form ahead of https://github.com/hashintel/hash/pull/1163 merging

### CODE REVIEW

File: packages/libs/error-stack/LICENSE-MIT.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 MIT License
```
Comment (nonparibus) [inline packages/libs/error-stack/LICENSE-MIT.md:1]:
```suggestion
# MIT License
```

File: sites/hashdev/src/_pages/blog/2_open-source.mdx [hunk 2]
```diff
@@ -15,13 +15,13 @@ At HASH, we publish as much of our code as possible in public.
 Depending on the code in question, it'll typically be released under one of three types of license, each of which has slightly different terms attached to it governing what you can and can't do with the underlying source:
 
 1.  **Open-source and permissive licenses:** in essence these licenses give people the freedom to do pretty much whatever they like, including extending the code and keeping it private (making this the most flexible category of license). The entirety of the [Block Protocol](https://github.com/blockprotocol/blockprotocol) is available under these terms (by way of the _MIT License_), alongside most of our blocks and connectors.
```
Comment (nonparibus) [inline sites/hashdev/src/_pages/blog/2_open-source.mdx:17]:
```suggestion
1.  **Open-source and permissive licenses:** in essence these licenses give people the freedom to do pretty much whatever they like, including extending the code and keeping it private (making this the most flexible category of license). The entirety of the [Block Protocol](https://github.com/blockprotocol/blockprotocol) is available under these terms (by way of the _MIT License_), alongside most of our blocks and connectors. Some of our packages like [`error-stack`](https://github.com/hashintel/hash/tree/main/packages/libs/error-stack) and [`deer`](https://github.com/hashintel/hash/tree/main/packages/libs/deer) are dual-licensed under multiple permissive OSS licenses (in their case both _MIT_ and _Apache 2.0_).
```

File: sites/hashdev/src/_pages/blog/5_build-blocks-in-any-language.mdx [hunk 3]
```diff
@@ -3,7 +3,7 @@ authorPhoto: blog/authors/ahmad-sattar.png
 postPhoto: blog/5_build-blocks-in-any-language/FSharp_Cover.jpg
 title: "How to build Block Protocol blocks in any language"
 subtitle: "Transpiling from F# to JavaScript to build a complex spreadsheet block"
-author: "Ahmad Sattar Atta"
+author: "Senior Ahmad Sattar Atta"
```
Comment (indietyp) [inline sites/hashdev/src/_pages/blog/5_build-blocks-in-any-language.mdx:6]: I think the "Senior" is placed in the wrong like.

Comment (nonparibus) [reply sites/hashdev/src/_pages/blog/5_build-blocks-in-any-language.mdx:6]: D'oh. Señor, maybe. Thanks for spotting this!

Comment (nonparibus) [reply sites/hashdev/src/_pages/blog/5_build-blocks-in-any-language.mdx:6]:
```suggestion
author: "Ahmad Sattar Atta"
```

File: sites/hashdev/src/_pages/blog/5_build-blocks-in-any-language.mdx [hunk 4]
```diff
@@ -3,7 +3,7 @@ authorPhoto: blog/authors/ahmad-sattar.png
...
 jobTitle: "Platform Engineer, HASH"
```
[the remark this answers is not in the document]
Comment (nonparibus) [reply sites/hashdev/src/_pages/blog/5_build-blocks-in-any-language.mdx]: Applied in https://github.com/hashintel/hash/pull/1172/commits/c7ed10a03682337ddfed00969813566177315844

File: packages/libs/error-stack/README.md [hunk 5]
```diff
@@ -97,6 +97,19 @@ This will most likely result in an error and print
 
 ![](https://github.com/hashintel/hash/blob/8ed55bd73045fba83a7ea2e199b31d5b829537b9/packages/libs/error-stack/assets/full.png?raw=true)
 
-Please see the [documentation] for a full description.
+## Usage
+
+Please see the [documentation].
+
+For more examples of `error-stack` in use, please check out the [examples](./examples) folder.
```
Comment (TimDiekmann) [inline packages/libs/error-stack/README.md:104]: I wonder if we want to refer to the examples folder by an absolute path as well, so the link is also working from external pages (like crates.io).

Comment (nonparibus) [reply packages/libs/error-stack/README.md:104]: @TimDiekmann Good shout.

```suggestion
For more examples of `error-stack` in use, please check out the [examples](https://github.com/hashintel/hash/tree/main/packages/libs/error-stack/examples) folder.
```

File: .github/ISSUE_TEMPLATE/bug-report-deer.yml [hunk 6]
```diff
@@ -0,0 +1,98 @@
+name: Bug Report (error-stack)
```
Comment (TimDiekmann) [inline .github/ISSUE_TEMPLATE/bug-report-deer.yml:1]:
```suggestion
name: Bug Report (deer)
```

File: .github/ISSUE_TEMPLATE/bug-report-deer.yml [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+description: Create a report to help us improve `error-stack`
```
Comment (TimDiekmann) [inline .github/ISSUE_TEMPLATE/bug-report-deer.yml:2]:
```suggestion
description: Create a report to help us improve `deer`
```

File: .github/ISSUE_TEMPLATE/bug-report-deer.yml [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+labels:
+  - A-error-stack
```
Comment (TimDiekmann) [inline .github/ISSUE_TEMPLATE/bug-report-deer.yml:4]:
```suggestion
  - A-deer
```

File: .github/ISSUE_TEMPLATE/bug-report-deer.yml [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+      placeholder: e.g. x86_64-unknown-linux-musl
+    validations:
+      required: false
+
+  - type: input
+    id: version
+    attributes:
+      label: Version
+      description: What error-stack version do you use? If you use the git version please also post the revision.
+      placeholder: e.g. v0.2.1
```
Comment (TimDiekmann) [inline .github/ISSUE_TEMPLATE/bug-report-deer.yml:79]: As we currently not publishing it, I'd change this to `revision`

```suggestion
    id: revision
    attributes:
      label: Revision
      description: What revision of `deer` do you use?
      placeholder: e.g. abcd1234
```

File: .github/ISSUE_TEMPLATE/bug-report-deer.yml [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+    validations:
+      required: false
+
+  - type: input
+    id: features
+    attributes:
+      label: Features
+      description: What features are enabled?
+      placeholder: default, spantrace
```
Comment (TimDiekmann) [inline .github/ISSUE_TEMPLATE/bug-report-deer.yml:88]:
```suggestion
      placeholder: default, std
```

**Comment by indietyp** [comment]
> Adds `deer` crate GitHub Issue bug report form ahead of https://github.com/hashintel/hash/pull/1163 merging

Does this PR block #1163?

File: sites/hashdev/src/_pages/blog/0002_open-source.mdx [hunk 11]
```diff
@@ -14,14 +14,14 @@ At HASH, we publish as much of our code as possible in public.
...
-1.  **Open-source and copyleft licenses:** these again give people the freedom to use the code however they like, provided that if they extend or modify the code they contribute those changes back into the open source as well under the same terms. Our [HASH](https://hash.ai/platform/hash) application is available under the _AGPL_ which guarantees this freedom.
+1.  **Open-source and permissive licenses:** in essence these licenses give people the freedom to do pretty much whatever they like, including extending the code and keeping it private (making this the most flexible category of license). The entirety of the [Block Protocol](https://github.com/blockprotocol/blockprotocol) is available under these terms (by way of the _MIT License_), alongside most of our blocks and connectors. Some of our packages like [`error-stack`](https://github.com/hashintel/hash/tree/main/packages/libs/error-stack) and [`deer`](https://github.com/hashintel/hash/tree/main/packages/libs/deer) are dual-licensed under multiple permissive OSS licenses (i.e. both the _MIT License_ and _Apache 2.0 License_).
```
Comment (TimDiekmann) [inline sites/hashdev/src/_pages/blog/0002_open-source.mdx:17]:
```suggestion
1.  **Open-source and permissive licenses:** in essence these licenses give people the freedom to do pretty much whatever they like, including extending the code and keeping it private (making this the most flexible category of license). The entirety of the [Block Protocol](https://github.com/blockprotocol/blockprotocol) is available under these terms (by way of the _MIT License_), alongside most of our blocks and connectors. Some of our packages like [`error-stack`](https://github.com/hashintel/hash/tree/main/packages/libs/error-stack) and [`deer`](https://github.com/hashintel/hash/tree/main/packages/libs/deer) are dual-licensed under multiple permissive OSS licenses (i.e. both the _MIT License_ and _Apache 2.0 License_) at your option.
```

**Review by TimDiekmann — CHANGES_REQUESTED** [review]

File: sites/hashdev/src/_pages/blog/0002_open-source.mdx (hunk 11 above)
Comment (nonparibus) [reply sites/hashdev/src/_pages/blog/0002_open-source.mdx:17]:
```suggestion
1.  **Open-source and permissive licenses:** in essence these licenses give people the freedom to do pretty much whatever they like, including extending the code and keeping it private (making this the most flexible category of license). The entirety of the [Block Protocol](https://github.com/blockprotocol/blockprotocol) is available under these terms (by way of the _MIT License_), alongside most of our blocks and connectors. Some of our packages like [`error-stack`](https://github.com/hashintel/hash/tree/main/packages/libs/error-stack) and [`deer`](https://github.com/hashintel/hash/tree/main/packages/libs/deer) are released under multiple permissive licenses (i.e. incorporable under both _MIT_ and _Apache 2.0_ licenses, at your option).
```

Comment (nonparibus) [reply sites/hashdev/src/_pages/blog/0002_open-source.mdx]: Elected for a slight rephrase instead, but thanks for flagging this.

Comment (TimDiekmann) [reply sites/hashdev/src/_pages/blog/0002_open-source.mdx]: Sounds good 👍🏻

**Comment by nonparibus** [comment]
> > Adds `deer` crate GitHub Issue bug report form ahead of #1163 merging
> 
> Does this PR block #1163?

We're fine with them merging in either order!

**Review by TimDiekmann — APPROVED** [review]
I updated the `README.md`s in `sites/hashdev/public/blog` to point to the correct folders in https://github.com/hashintel/hash/pull/1172/commits/674e2740abe3dd6cc7403ba227a16b72a0518db8

Changes are looking good, but I noticed, that a readme for the 5. and 9. blog post are missing.

**Review by TimDiekmann — DISMISSED** [review]

**Review by TimDiekmann — APPROVED** [review]

**Pull request merged**